### PR TITLE
Basic throttling for nginx updates

### PR DESCRIPTION
**Pull request by chbatey**
If many updates came in in a short space of time nginx spawns
many workers processes resulting in OOM kills

### CODE REVIEW

File: cmd/feed-ingress/main.go [hunk 1]
```diff
@@ -17,39 +17,26 @@ import (
...
-	elbLabelValue                     string
-	elbRegion                         string
-	elbExpectedNumber                 int
-	pushgatewayURL                    string
-	pushgatewayIntervalSeconds        int
-	pushgatewayLabels                 cmd.KeyValues
-	accessLog                         bool
-	accessLogDir                      string
-	nginxLogHeaders                   string
+	debug                      bool
```
Comment (jsravn) [inline cmd/feed-ingress/main.go]: Nice

File: nginx/nginx.go [hunk 2]
```diff
@@ -202,6 +231,23 @@ func (lb *nginxLoadBalancer) periodicallyUpdateMetrics() {
 	}
 }
 
+func (lb *nginxLoadBalancer) signaler() {
```
Comment (jsravn) [inline nginx/nginx.go]: Can we improve the function name, signaller is the thing that signals nginx and we already use that name. This is the loop.

File: nginx/nginx.go [hunk 3]
```diff
@@ -70,15 +73,36 @@ func (s *osSignaller) sighup(p *os.Process) error {
...
+	lb.updateRequired.Lock()
+	defer lb.updateRequired.Unlock()
+	lb.updateRequired.update = true
+}
+
+func (lb *nginxLoadBalancer) signalIfRequired() {
+	lb.updateRequired.Lock()
+	defer lb.updateRequired.Unlock()
+	if lb.updateRequired.update {
+		lb.signaller.sighup(lb.cmd.Process)
```
Comment (jsravn) [inline nginx/nginx.go]: Can we log when it gets updated?

File: nginx/nginx.go [hunk 4]
```diff
@@ -221,25 +267,22 @@ func (lb *nginxLoadBalancer) Stop() error {
...
 	}
 
 	if updated {
-		err = lb.signaller.sighup(lb.cmd.Process)
-		if err != nil {
-			return fmt.Errorf("unable to signal nginx to reload: %v", err)
-		}
-		log.Info("Nginx updated")
+		lb.signalRequired()
+		log.Infof("Nginx update required (can take up to %d seconds)", lb.Conf.UpdateFrequencySeconds)
```
Comment (jsravn) [inline nginx/nginx.go]: Not sure this is worthwhile logging - if we put a log message when it does the actual update. We'll already get lots of log spam when an update comes in.

File: nginx/nginx.go (hunk 2 above)
Comment (chbatey) [inline nginx/nginx.go]: done

File: nginx/nginx.go (hunk 3 above)
Comment (chbatey) [inline nginx/nginx.go]: yep

File: nginx/nginx.go (hunk 4 above)
Comment (chbatey) [inline nginx/nginx.go]: removed

**Pull request merged**